### DISCUSSION TITLE
Discussion: Steam crashes with Decky

### DISCUSSION THREAD

**Discussion by cimba007** [SteamDeckHomebrew/decky-loader#166] [Q&A] [answered]
Recently I noticed a lot of UI crashes combined with delayed OnscreenKeyboard (2-3 Sec delay).

For now I am not sure if this is the fault of decky or CEF-Remotedebugging.

Is anybody experiencing something similiar?

Steamdeck on Stable
Decky on v2.0.5-pre17
Plugins - none

**Comment by TrainDoctor** [comment]
Could you please include logs by using ``journalctl -b0 -u plugin_loader > loader.log``, which you can then copy from the ``/home/deck/`` directory on your Steam Deck. Don't copy paste the file just go ahead and upload it somewhere like pastebin etc.

Also just in case, can you disable/uninstall decky and try to replicate the issues you've been encountering?

**Comment by cimba007** [comment]
https://pastebin.com/96HLg2n2

To specify it a little bit .. the "crash" can be provoked if you are ingame and press steam-button + x until the ui becomes more and more unresponsive followed by a steam-ui crash or hard steamdeck reboot.

I will now try to uninstall decky and leave debug/dev-mode enabled.

Testcase 1: Devmode + CEF-Debugging enabled without decky = Keyboard Popup responsive
Testcase 2: Devmode + CEF-Debugging enabled with decky = Keyboard Popup unresponsive
       For now I cant force a crash but after 5seconds spamming the UI won't responde for 30 seconds at all
Testcase 3: Devmode enabled, CEF-Debugging disabled with decky installed = Keyboard Popup responsive

**Comment by jccr** [comment]
I also experienced this issue while opening the OSK while in a game, which led me to here. I'll try to get a log capture.

**Comment by failzers** [comment]
Been experiencing the same issue for weeks, logs have driven me crazy. Sadly though, I've still encountered the same issues whilst having Decky uninstalled. I imagine gamescope must be currently broken, bringing the whole system down. Anyone with this issue who seems to also crash if you have the performance overlay on and just play whatever app/game without even opening the menus? (though that seems to force a crash too sometimes)

**Comment by TrainDoctor** [comment]
This increasingly seems like an issue with no connection to decky-loader. Moving this to a proper discussion.

**Reply by cimba007** [reply]
@TrainDoctor This increasingly seems to be a real issue. It's there still nothing we can provide for further analysis?

**Reply by TrainDoctor** [reply]
> @TrainDoctor This increasingly seems to be a real issue. It's there still nothing we can provide for further analysis?

This is not a decky loader issue. It's due to a few issues with gamescope, the Wayland compositor valve uses.
Our tool only surfaced the issue. I'd appreciate some faith that I'm not talking out of my ass.
https://github.com/Plagman/gamescope/issues/613#issuecomment-1248903453

**Reply by cimba007** [reply]
Ty for reference.

**Comment by cimba007** [comment]
@jccr you got the problem with decky running?
@TrainDoctor I tried 3 tests and I am pretty confident it only happens with decky installed and loaded.

**Reply by failzers** [reply]
Have you tried uninstalling decky and just turning on level 4 of the performance overlay? That seems to increase the likelihood of a crash for me.

**Reply by jccr** [reply]
I started observing the problems after installing the latest decky prerelease, and the issue went away after disabling CEF debugging.

**Reply by jccr** [reply]
> Have you tried uninstalling decky and just turning on level 4 of the performance overlay? That seems to increase the likelihood of a crash for me. 

I think I was using level 4 around the time of the crashes.. interesting point

**Reply by cimba007** [reply]
If you disable CEF you disable decky dont you?
@failzers I am unable to trigger any unresponsiveness in the OSK with perf. level 4 .. no crashes.

**Reply by jccr** [reply]
> If you disable CEF you disable decky dont you?
> @failzers I am unable to trigger any unresponsiveness in the OSK with perf. level 4 .. no crashes.

Yea, I am disabling Decky by disabling CEF debugging. I'm playing RuneScape via RuneLite and I want to stop the crashes with the OSK for now 😅
I'll enable Decky by enabling CEF debugging again, go into another gaming session (without perf lv4) and see if I get the OSK crashing my steam client.

**Reply by infernn** [reply]
> > If you disable CEF you disable decky dont you?
> > @failzers I am unable to trigger any unresponsiveness in the OSK with perf. level 4 .. no crashes.
> 
> Yea, I am disabling Decky by disabling CEF debugging. I'm playing RuneScape via RuneLite and I want to stop the crashes with the OSK for now 😅 I'll enable Decky by enabling CEF debugging again, go into another gaming session (without perf lv4) and see if I get the OSK crashing my steam client.

are  you still having crash after you have disabled decky? i'm having reboot with screen that say "verify installation" when i close a game.

**Comment by cimba007** [comment]
@TrainDoctor Can you confirm that with CEF+Decky enabled the OSK is responsive (Steam+X) within different situations?
I am not talking about the OSK performance while typing.. only the delay it takes for the OSK to pop up.

**Reply by cimba007** [reply]
One thing to add maybe .. I experience my problems while playing GW2. Today I enabled Deckly after some days without it and .. guess it .. 3rd keyboard open = steam crash

**Comment by cimba007** [comment]
Still present with [Prerelease v2.1.0-pre1]

**Comment by cimba007** [comment]
Just for reference https://v.redd.it/rqdziicis8o91

**Comment by lbibass** [comment]
I also get crashes in a variety of games and emulators when Decky is enabled. Happens in portal, dolphin, yuzu, etc. 

I click the Steam button, or the … button, and the screen goes black and my Steam Deck reboots.

**Comment by lbibass** [comment]
Update: I’ll get crashes just navigating around the OS. using Main OS, Steam Beta. 

This makes decky unusable, which is a major shame given that I need it for proper emulation performance.

**Reply by TrainDoctor** [reply]
https://github.com/Plagman/gamescope/issues/613#issuecomment-1248903453

**Reply by dan3093** [reply]
@TrainDoctor is that last comment on that thread a fix? Is it going to be released in an update soon?

**Reply by TrainDoctor** [reply]
> @TrainDoctor is that last comment on that thread a fix? Is it going to be released in an update soon?

Please read the thread, it's up to valve to fix the issues Josh mentioned and then the issue will no longer be an issue.

**Comment by infernn** [comment]
i'm having system reboot sometimes after i close a game with a screen that say "verify installation" is problem related to decky?

**Reply by TrainDoctor** [reply]
I can't say for sure but here's a link to the relevant issues that are being surfaced by decky loader:
https://github.com/Plagman/gamescope/issues/613#issuecomment-1248903453

**Comment by TrainDoctor** [comment, marked as answer]
Converted to Q&A and including this link to be marked as the answer:
https://github.com/Plagman/gamescope/issues/613#issuecomment-1248903453
